src/openrct2/ride/TrackDesign.cpp

```diff
@@ -1234,40 +1240,40 @@ static bool TrackDesignPlaceSceneryElement(
                     }
                     footpath_connect_edges(mapCoord, reinterpret_cast<TileElement*>(pathElement), flags);
                     footpath_update_queue_chains();
-                    return true;
+                    return totalCost;
                 }
                 break;
             default:
                 _trackDesignPlaceStateSceneryUnavailable = true;
-                return true;
+                return totalCost;
         }
-        _trackDesignPlaceCost = add_clamp_money32(_trackDesignPlaceCost, cost);
+        totalCost = add_clamp_money32(totalCost, cost);
         if (_trackDesignPlaceOperation != PTD_OPERATION_PLACE)
         {
             if (cost == MONEY32_UNDEFINED)
             {
-                _trackDesignPlaceCost = MONEY32_UNDEFINED;
+                totalCost = MONEY32_UNDEFINED;
             }
         }
-        if (_trackDesignPlaceCost != MONEY32_UNDEFINED)
+        if (totalCost != MONEY32_UNDEFINED)
         {
-            return true;
+            return totalCost;
         }
         if (_trackDesignPlaceOperation == PTD_OPERATION_PLACE)
         {
-            return true;
+            return totalCost;
         }
-        return false;
+        return std::nullopt;
     }
-    return true;
+    return totalCost;
 }
 
 /**
  *
  *  rct2: 0x006D0964
  */
-static int32_t track_design_place_all_scenery(
-    const std::vector<TrackDesignSceneryElement>& sceneryList, const CoordsXYZ& origin)
+static std::optional<money32> track_design_place_all_scenery(
```

Comment:
Same would go for this function. Remove the passing in of the track cost (You have inadvertently implemented as per my suggestion as L1305 is meant to be `+=`)